src/libbson/src/bson/bson-macros.h

```diff
@@ -196,18 +196,48 @@
          abort ();                                         \
       }                                                    \
    } while (0)
-       
+
+/**
+ * @brief Assert the expression `Assertion`, and evaluates to `Value` on
+ * success.
+ */
+#define BSON_ASSERT_INLINE(Assertion, Value)                              \
+   ((void) ((Assertion) ? (0)                                             \
+                        : ((fprintf (stderr,                              \
+                                     "%s:%d %s(): Assertion '%s' failed", \
+                                     __FILE__,                            \
+                                     __LINE__,                            \
+                                     BSON_FUNC,                           \
+                                     #Assertion),                         \
+                            abort ()),                                    \
```

Comment:
Since we're calling abort() rather than exit(), does the abbove fprintf() need fflush()?

Reply:
Whether files are flushed is implementation-defined, it appears. In all of my testing with the various `assert` macros, I've not found that files are not implicitly flushed before the program terminates. None of the other assertion macros flush, so I have omitted that here as well.

Reply:
+1, it does appear that this is implementation-defined (and currently at least glibc does not appear to flush specifically because of contention issues, although it has done so in the past).